app/frontend/src/components/designer/FormViewer.vue

```
@@ -354,27 +339,69 @@ export default {
 
+    // FormIO submit event
     // eslint-disable-next-line no-unused-vars
     async onSubmit(submission) {
       if (this.preview) {
         alert('Submission disabled during form preview');
         return;
       }
+
+      const errors = await this.doSubmit(submission);
+
       // if we are here, the submission has been saved to our db
       // the passed in submission is the formio submission, not our db persisted submission record...
       // fire off the submitDone event.
       // console.info(`onSubmit(${JSON.stringify(submission)})`) ; // eslint-disable-line no-console
-      this.currentForm.events.emit('formio.submitDone');
+      if (errors) {
+        this.addNotification({
+          message: errors,
+          consoleError: `Error submiting the form: ${errors}`,
+        });
+      } else {
+        this.currentForm.events.emit('formio.submitDone');
+      }
+    },
+
+    // Not a formIO event, our saving routine to POST the submission to our API
+    async doSubmit(submission) {
+      // console.info(`doSubmit(${JSON.stringify(submission)})`) ; // eslint-disable-line no-console
+      // since we are not using formio api
+      // we should do the actual submit here, and return any error that occurrs to handle in the submit event
+      let errMsg = undefined;
+      try {
+        const response = await this.sendSubmission(false, submission);
+
+        if ([200, 201].includes(response.status)) {
+          // all is good, flag no errors and carry on...
+          // store our submission result...
+          this.submissionRecord = Object.assign(
+            {},
+            this.submissionId ? response.data.submission : response.data
+          );
+          // console.info(`doSubmit:submissionRecord = ${JSON.stringify(this.submissionRecord)}`) ; // eslint-disable-line no-console
+        } else {
+          // console.error(response); // eslint-disable-line no-console
+          errMsg = 'An error occurred submitting this form';
```

This else block should throw an error instead of repeating the same errMsg assignment - then it'd be using the same catch block logic.

---

Indeed 👍